python/ray/monitor.py

```diff
@@ -88,6 +88,28 @@ def psubscribe(self, pattern):
         """
         self.primary_subscribe_client.psubscribe(pattern)
 
+    def handle_resource_demands(self, resource_load_by_shape):
+        """Handle the message.resource_load_by_shape protobuf for the demand
+        based autoscaling. Catch and log all exceptions so this doesn't
+        interfere with the utilization based autoscaler until we're confident
+        this is stable.
+
+        Args:
+            resource_load_by_shape (pb2.gcs.ResourceLoad): The resource demands
+                in protobuf form or None.
+        """
+        try:
+            if not self.autoscaler:
+                return
+            bundles = []
+            for resource_demand_pb in list(
+                    resource_load_by_shape.resource_demands):
+                request_shape = dict(resource_demand_pb.shape)
+                bundles.append(request_shape)
```

Comment:
Also I didn't see this issue before, but you need to take into account the queue size. For example, if here are N requests of this shape, push N shapes into the request, not just 1.